contracts/managers/fee/FeeManager.sol

```diff
@@ -139,108 +221,241 @@ contract FeeManager is AccessManaged {
+    function _updateFees(
+        RecipientFee[] calldata recipientFees,
+        FeeRecipientData storage feeData,
+        uint256 daoFee,
+        address feeAccount,
+        FeeType feeType
+    ) internal {
+        uint256 totalFee = daoFee;
+    
+        address[] memory oldRecipients = feeData.recipientAddresses;
+        for (uint256 i = 0; i < oldRecipients.length; i++) {
+            delete feeData.recipientFees[oldRecipients[i]];
+        }
+        
+        delete feeData.recipientAddresses;
+        
+        address[] memory newRecipients = new address[](recipientFees.length);
+        uint256[] memory newFees = new uint256[](recipientFees.length);
+
+        for (uint256 i = 0; i < recipientFees.length; i++) {
+            if (recipientFees[i].recipient == address(0)) {
+                revert InvalidFeeRecipientAddress();
+            }
+            
+            newRecipients[i] = recipientFees[i].recipient;
+            newFees[i] = recipientFees[i].feeValue;
+            
+            feeData.recipientFees[recipientFees[i].recipient] = recipientFees[i].feeValue;
+            totalFee += recipientFees[i].feeValue;
+        }
 
-        PlasmaVaultGovernance(PLASMA_VAULT).configurePerformanceFee(PERFORMANCE_FEE_ACCOUNT, newPerformanceFee);
-        plasmaVaultPerformanceFee = newPerformanceFee;
+        feeData.recipientAddresses = newRecipients;
 
-        emit PerformanceFeeUpdated(newPerformanceFee);
+        if (feeType == FeeType.MANAGEMENT) {
+            PlasmaVaultGovernance(PLASMA_VAULT).configureManagementFee(feeAccount, totalFee);
+            plasmaVaultTotalManagementFee = totalFee;
+            emit ManagementFeeUpdated(totalFee, newRecipients, newFees);
+        } else {
+            PlasmaVaultGovernance(PLASMA_VAULT).configurePerformanceFee(feeAccount, totalFee);
+            plasmaVaultTotalPerformanceFee = totalFee;
+            emit PerformanceFeeUpdated(totalFee, newRecipients, newFees);
+        }
+    }
+    
+    /// @notice Gets the management fee for a specific recipient
+    /// @param recipient The address of the recipient
+    /// @return The management fee value
+    function getManagementFee(address recipient) external view returns (uint256) {
+        return _managementFeeRecipientData.recipientFees[recipient];
     }
 
-    /// @notice Updates the management fee and reconfigures it in the PlasmaVaultGovernance contract.
-    /// @param managementFee_ The new management fee to be added to the DAO management fee.
-    function updateManagementFee(uint256 managementFee_) external restricted {
-        harvestManagementFee();
-
-        uint256 newManagementFee = managementFee_ + IPOR_DAO_MANAGEMENT_FEE;
+    /// @notice Gets all management fee recipient addresses
+    /// @return Array of recipient addresses
+    function getManagementFeeRecipients() external view returns (address[] memory) {
+        return _managementFeeRecipientData.recipientAddresses;
```

Comment:
Maybe it would be better to also return the fee values for each address, because at the moment it seems that this cannot be read from the contract.

Reply:
You can read using getManagementFee method

Reply:
I added one method for recipient and his fee per fee type (management, performance)